File: Server/src/main/content/global/skill/production/cooking/recipe/stew/CurryRecipe.java
package content.global.skill.production.cooking.recipe.stew;

import content.global.skill.production.cooking.recipe.Recipe;
import core.game.interaction.NodeUsageEvent;
import core.game.node.entity.player.Player;
import core.game.node.item.Item;

/**
 * The Curry recipe.
 */
public class CurryRecipe extends Recipe {

    private static final Item UNCOOKED_CURRY = new Item(2009);

    private static final Item UNCOOKED_STEW = new Item(2001);

    private static final Item SPICE = new Item(2007);

    private static final Item CURRY_LEAF = new Item(5970);

    @Override
    public void mix(Player player, NodeUsageEvent event) {
        if (event.getBaseItem().getId() == CURRY_LEAF.getId() || event.getUsedItem().getId() == CURRY_LEAF.getId()) {
            Item stew = event.getBaseItem().getId() == UNCOOKED_STEW.getId() ? event.getBaseItem() : event.getUsedItem();
            if (stew.getCharge() == 1000) {
                stew.setCharge(1);
            }
            int charge = stew.getCharge();
            if (charge < 3) {
                player.getInventory().remove(CURRY_LEAF);
                stew.setCharge(charge + 1);
            } else {
                if (player.getInventory().remove(stew) && player.getInventory().remove(CURRY_LEAF)) {
                    player.getInventory().add(UNCOOKED_CURRY);
                }
            }
            return;
        }
        if (player.getInventory().remove(event.getBaseItem()) && player.getInventory().remove(event.getUsedItem())) {
            player.getInventory().add(getProduct());
        }
    }

    @Override
    public Item getBase() {
        return UNCOOKED_STEW;
    }

    @Override
    public Item getProduct() {
        return UNCOOKED_CURRY;
    }

    @Override
    public Item[] getIngredients() {
        return new Item[]{SPICE, CURRY_LEAF};
    }

    @Override
    public Item[] getParts() {
        return new Item[]{UNCOOKED_STEW};
    }

    @Override
    public String getMixMessage(NodeUsageEvent event) {
        return "You mix the spice with the stew.";
    }

    @Override
    public boolean isSingular() {
        return true;
    }

}

File: Emulator/src/main/content/global/skill/cooking/type/stew/CurryRecipe.kt
package content.global.skill.cooking.type.stew

import content.global.skill.cooking.type.Recipe
import org.rs.consts.Items
import core.api.removeItem
import core.game.interaction.NodeUsageEvent
import core.game.node.entity.player.Player
import core.game.node.item.Item

/**
 * Curry recipe.
 */
class CurryRecipe : Recipe() {

    override fun mix(player: Player, event: NodeUsageEvent) {
        if (event.baseItem.id == CURRY_LEAF.id || event.usedItem.id == CURRY_LEAF.id) {
            val stew = if (event.baseItem.id == UNCOOKED_STEW.id) event.baseItem else event.usedItem
            if (stew.charge == 1000) {
                stew.charge = 1
            }
            val charge = stew.charge
            if (charge < 3) {
                player.inventory.remove(CURRY_LEAF)
                stew.charge = charge + 1
            } else {
                if (removeItem(player, stew) && removeItem(player, CURRY_LEAF)) {
                    player.inventory.add(UNCOOKED_CURRY)
                }
            }
            return
        }
        if (removeItem(player, event.baseItem) && removeItem(player, event.usedItem)) {
            player.inventory.add(product)
        }
    }

    override fun getBase(): Item {
        return UNCOOKED_STEW
    }

    override fun getProduct(): Item {
        return UNCOOKED_CURRY
    }

    override fun getIngredients(): Array<Item> {
        return arrayOf(SPICE, CURRY_LEAF)
    }

    override fun getParts(): Array<Item> {
        return arrayOf(UNCOOKED_STEW)
    }

    override fun getMixMessage(event: NodeUsageEvent): String {
        return "You mix the spice with the stew."
    }

    override fun isSingular(): Boolean {
        return true
    }

    companion object {
        private val UNCOOKED_CURRY = Item(Items.UNCOOKED_CURRY_2009)
        private val UNCOOKED_STEW = Item(Items.UNCOOKED_STEW_2001)
        private val SPICE = Item(Items.SPICE_2007)
        private val CURRY_LEAF = Item(Items.CURRY_LEAF_5970)
    }
}
